examples/box_game/box_game_spectator.rs

```diff
@@ -60,45 +61,51 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     // continue building/running the app like you normally would
     app.insert_resource(Msaa { samples: 4 })
-        .insert_resource(WindowDescriptor {
-            width: 720.,
-            height: 720.,
-            title: "GGRS Box Game".to_owned(),
-            ..Default::default()
-        })
         .insert_resource(opt)
         .add_plugins(DefaultPlugins)
         .add_startup_system(setup_system)
         // add your GGRS session
-        .insert_resource(sess)
-        .insert_resource(SessionType::SpectatorSession)
+        .insert_resource(Session::SpectatorSession(sess))
         // register a resource that will be rolled back
         .insert_resource(FrameCount { frame: 0 })
         //print some network stats - not part of the rollback schedule as it does not need to be rolled back
-        .insert_resource(NetworkStatsTimer(Timer::from_seconds(2.0, true)))
+        .insert_resource(NetworkStatsTimer(Timer::from_seconds(
+            2.0,
+            TimerMode::Repeating,
+        )))
         .add_system(print_network_stats_system)
         .add_system(print_events_system)
         .run();
 
     Ok(())
 }
 
-fn print_events_system(mut session: ResMut<SpectatorSession<GGRSConfig>>) {
-    for event in session.events() {
-        println!("GGRS Event: {:?}", event);
+fn print_events_system(mut session: ResMut<Session<GGRSConfig>>) {
+    match session.as_mut() {
+        Session::P2PSession(s) => {
+            for event in s.events() {
+                println!("GGRS Event: {:?}", event);
+            }
+        }
+        _ => panic!("This examples focus on p2p."),
```

Comment:
ditto

Reply:
There's also a indication of the crash encountered by @gschup : this example is not focusing on "p2p" (well it is, but it's a spectator session...) so this sentence and the match is not correct.